plugins/net_plugin/net_plugin.cpp

```diff
@@ -1171,6 +1180,7 @@ namespace eosio {
          controller& cc = my_impl->chain_plug->chain();
          signed_block_ptr sb = cc.fetch_block_by_number( num );
          if( sb ) {
+            c->last_block = num;
```

Comment:
It looks like you will set last_block on sending and receiving. It might be useful to have both, but not one you have to guess which it is.